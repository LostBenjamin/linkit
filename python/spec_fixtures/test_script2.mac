# red green and blue wandering pixels with mirror

#[name {number}] start macro
#(name) run macro

# variables
$max 90
$max-half 45
$render-time 50
$sequence-time 20
$fade-type ffd
#$red-seq 0
#$green-seq 1
#$blue-seq 2
$wander-range 3
$wander-neg-range -1

# use variables or recall macro numbers <name>

# command value substitutions
# these should be all caps built-in?
$get-current -1
$macro-seq -4

# fancy app
[fancy 10]
(init)
(setup-rendering)
(start-components)

# initialization
[init]
app

# rendering
[setup-rendering]
<render-time>,<render>:sch

[render]
mir
flu

# start up components
[start-components]
(start-red)
(start-green)
(start-blue)

# red wanderer
[start-red]
{red-seq},<max>,0:seq
<sequence-time>,<red-sequence>:sch

[red-sequence]
<red-seq>,<macro-seq>,<red-math>:seq
pos
<red-seq>,0,1:pal
<fade-type>
rst

[red-math]
<wander-range>:rng:psh
<wander-neg-range>:psh
add
<red-seq>,<get-current>:seq:psh
add
<max-half>:psh
mod

# green wanderer
[start-green]
{green-seq},<max>,0:seq
<sequence-time>,<green-sequence>:sch

[green-sequence]
<green-seq>,<macro-seq>,<green-math>:seq
pos
<green-seq>,0,1:pal
<fade-type>
rst

[green-math]
<wander-range>:rng:psh
<wander-neg-range>:psh
add
<green-seq>,<get-current>:seq
psh
add
<max-half>:psh
mod

# blue wanderer
[start-blue]
{blue-seq},<max>,0:seq
<sequence-time>,<blue-sequence>:sch

[blue-sequence]
<blue-seq>,<macro-seq>,<blue-math>:seq
pos
<blue-seq>,0,1:pal
<fade-type>
rst

[blue-math]
<wander-range>:rng:psh
<wander-neg-range>:psh
add
<blue-seq>,<get-current>:seq
psh
add
<max-half>:psh
mod

