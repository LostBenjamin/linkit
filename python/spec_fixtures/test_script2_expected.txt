10:set:22:run:23:run:26:run
11:set:2,-4,27:seq:pos:2,0,1:pal:ffd:rst
13:set:3:rng:psh:-1:psh:add:0,-1:seq:psh:add:45:psh:mod
15:set:mir:flu
16:set:0,-4,13:seq:pos:0,0,1:pal:ffd:rst
18:set:3:rng:psh:-1:psh:add:1,-1:seq:psh:add:45:psh:mod
20:set:2,90,0:seq:20,11:sch
21:set:1,90,0:seq:20,24:sch
22:set:app
23:set:50,15:sch
24:set:1,-4,18:seq:pos:1,0,1:pal:ffd:rst
26:set:29:run:21:run:20:run
27:set:3:rng:psh:-1:psh:add:2,-1:seq:psh:add:45:psh:mod
29:set:0,90,0:seq:20,16:sch

